src/libs/parsers/tests/formatDates.test.ts

```diff
@@ -8,86 +8,368 @@ import {
+    const grouped: GroupResult = {}
+    const expected = {
+      formatDates: [],
+      arrayDays: [],
+    }
+    const result = formatGroupedDates(grouped)
+    expect(result.formatDates).toEqual(expected.formatDates)
+    expect(result.arrayDays).toEqual(expected.arrayDays)
+  })
+})
+
+describe('getFormattedDates', () => {
+  beforeAll(() => {
+    mockdate.set(NOVEMBER_1_2020)
+  })
+
+  it('should return undefined when undefined or empty array is given', () => {
+    expect(getFormattedDates(undefined)).toEqual(undefined)
+    expect(getFormattedDates([])).toEqual(undefined)
+  })
+
+  it('should return only future dates', () => {
+    expect(
+      getFormattedDates([OCTOBER_5_2020.toISOString(), NOVEMBER_12_2020.toISOString()])
+    ).toEqual(formatToFrenchDate(NOVEMBER_12_2020))
+  })
+
+  it('should return undefined if the dates array is invalid', () => {
+    const dates = ['invalid date']
+    const result = getFormattedDates(dates)
+    expect(result).toEqual(undefined)
+  })
+
+  it('should return a formatted date if there is only one unique dates', () => {
+    const result = getFormattedDates([DECEMBER_14_2020.toISOString()])
+    expect(result).toEqual('14 décembre 2020')
+  })
+
+  it('should return a formatted string of 1 date if there are two similar date ', () => {
+    const result = getFormattedDates([
+      DECEMBER_14_2020.toISOString(),
+      DECEMBER_14_2020.toISOString(),
+    ])
+    expect(result).toEqual('14 décembre 2020')
+  })
+
+  it('should return a formatted string date of two dates with the same month', () => {
+    const result = getFormattedDates([
+      NOVEMBER_12_2020.toISOString(),
+      NOVEMBER_13_2020.toISOString(),
+    ])
+    expect(result).toEqual('les 12 et 13 novembre 2020')
+  })
+
+  it('should return a formatted string date of 1 dates if there is two same dates', () => {
+    const result = getFormattedDates([
+      NOVEMBER_20_2020_MORNING.toISOString(),
+      NOVEMBER_20_2020_EVENING.toISOString(),
+    ])
+    expect(result).toEqual('le 20 novembre 2020')
+  })
+
+  it('should return a formatted string date of 2 dates with the same day, smae month, but different years', () => {
+    const result = getFormattedDates([
+      NOVEMBER_13_2020.toISOString(),
+      NOVEMBER_13_2021.toISOString(),
+    ])
+    expect(result).toEqual('le 13 novembre 2020 et le 13 novembre 2021')
+  })
+
+  it('should return a formatted string date that includes three dates with different months', () => {
+    const result = getFormattedDates([
+      AUGUST_5_2021.toISOString(),
+      SEPTEMBER_5_2021.toISOString(),
+      OCTOBER_5_2021.toISOString(),
+    ])
+    expect(result).toEqual('le 5 août 2021, le 5 septembre 2021 et le 5 octobre 2021')
+  })
+
+  it('should return a formatted string date that includes three dates, with two dates in different months of the same year and one dates in a different month and in a different year', () => {
```

Comment:
`should return a formatted string date including three dates, with two dates of different month in the same year and one date of different month and year`